libarchive/adapters/archive_read.py

```diff
@@ -49,6 +49,14 @@ def _archive_read_support_format_all(archive):
         message = c_archive_error_string(archive)
         raise libarchive.exception.ArchiveError(message)
 
+def _archive_read_add_passphrase(archive, passphrase):
+    try:
```

Comment:
Encode passphrase to bytes first: `passphrase.encode('utf-8')`

Reply:
That will cause Python 3 to error ...